chain/vm/runtime.go

```diff
@@ -152,13 +154,19 @@ func (rt *Runtime) GetRandomness(personalization crypto.DomainSeparationTag, ran
 	return res
 }
 
+func (rs *Runtime) InternalCallSeqNum() int64 {
+	return int64(len(rs.internalExecutions)) + rs.callDepth
```

Comment:
I don't think this will do the thing, the internalExecutions are per call, as in, they only track subcalls of this particular invocation (i made it recursive lately, but thats not accounted for here).

So, if A calls B, B calls C, D, E, then returns, the A calls F, the InternalCallSeqNum will be 3 (ish) which will be the same number that B sees after its call to C. Or something like that.

Reply:
Yeah, we cannot use `+` in there as calDepth is not monotonically increasing.
Can we just use a counter increasing on every send?

Reply:
Ach, sorry, I was trying to be clever, and convinced myself this worked.

Yeah, I'll just add a counter for callSeqNum...Shall we keep callDepth, though? It might have utility later.